Document the AllowWorkloadDisruption migration option (#863)
kubevirt/user-guide: docs/compute/live_migration.md

fails to converge within the specified `AcceptableCompletionTime`. This option
provides a balance between ensuring migration completion and mitigating risks
associated with different migration strategies.

#### Understanding the AllowWorkloadDisruption

The behavior dictated by `AllowWorkloadDisruption` depends on its configuration:

- **When disabled**:
  - Workloads are not disrupted during migration.
  - If the migration does not converge within the `AcceptableCompletionTime`, it will be canceled and may need to be restarted.
  - This approach minimizes workload disruption, however, "busy" workloads may not migrate in time, which increases the risk of repeated migration.

- **When enabled**:
  - The migration controller is allowed to disrupt the workload associated with long-running migrations to facilitate migration completion.
  - When `AcceptableCompletionTime` threshold is reached, the migration controller will switch the migration into a post-copy
    mode. If post-copy is not permitted, the migration controller will instead pause the corresponding Virtual Machine Instance to ensure migration
    completion.

> #### Configuration

Author: I did, it's just down below: 

```
Bear in mind that most of these configuration can be overridden and fine-tuned to
a specified group of VMs. For more information, please see [Migration Policies](../cluster_admin/migration_policies.md).

```

To configure `AllowWorkloadDisruption`:

1. **Disabling Workload Disruption**:
   - Leave `AllowWorkloadDisruption` unset or explicitly set it to `false`.

>    - Example:
>      ```yaml
>     spec:

Reviewer: Needs a newline between colon and codeblock
```suggestion
   - Example:
   
     ```yaml
    spec:
```

      configuration:
        migrations:
         allowWorkloadDisruption: false
     ```

2. **Enabling Workload Disruption**:
   - Set `AllowWorkloadDisruption` to `true` to allow the migration controller to act after the `AcceptableCompletionTime` threshold.

>    - To enable post-copy, set `AllowPostCopy` to `true`. Otherwise, if paused mode is prefered, unset `AllowPostCopy` or explicitly set it to `false`:
>      ```yaml

Reviewer: ```suggestion
   - To enable post-copy, set `AllowPostCopy` to `true`. Otherwise, if paused mode is prefered, unset `AllowPostCopy` or explicitly set it to `false`:
   
     ```yaml
```

    spec:
      configuration:
        migrations:
         allowWorkloadDisruption: true
         allowPostCopy: true
     ```

#### Key Considerations

- `AllowWorkloadDisruption` determines whether the migration controller can prioritize completing the migration over avoiding workload disruption.

> - Post-copy migrations, when enabled, poses some risk of data loss if a failure occurs during the post-copy phase.

Reviewer: ```suggestion
- Post-copy migration, when enabled, poses some risk of data loss if a failure occurs during the post-copy phase.
```
Or drop the 's' from 'poses'

> - Pausing the workload facilitates migration completion with no risk of data loss but may result in a temporary workload inactivity.

Reviewer: ```suggestion
- Pausing the workload facilitates completion of the migration without risk of data loss but may result in temporary workload inactivity.
```

The behavior dictated by `AllowWorkloadDisruption` depends on its configuration:

- **When disabled**:
  - Workloads are not disrupted during migration.
  - If the migration does not converge within the `AcceptableCompletionTime`, it will be canceled and may need to be restarted.
  - This approach minimizes workload disruption, however, "busy" workloads may not migrate in time, which increases the risk of repeated migration.

- **When enabled**:
  - The migration controller is allowed to disrupt the workload associated with long-running migrations to facilitate migration completion.
  - When `AcceptableCompletionTime` threshold is reached, the migration controller will switch the migration into a post-copy
    mode. If post-copy is not permitted, the migration controller will instead pause the corresponding Virtual Machine Instance to ensure migration
    completion.

#### Configuration

To configure `AllowWorkloadDisruption`:

1. **Disabling Workload Disruption**:
   - Leave `AllowWorkloadDisruption` unset or explicitly set it to `false`.
   - Example:

     ```yaml
    spec:
      configuration:
        migrations:
         allowWorkloadDisruption: false

>      ```

Reviewer 2: Not sure why, but this block doesn't display properly, same for the next one.

Author: yup, no clue :)

Reviewer: Looks like it's the indentation. I thought it was the lack of newline but you don't need it if you have exactly 4 spaces of indentation (it currently has 5), which will keep it in line with the numbered list element. 
You could also just have no spaces as the difference isn't particularly noticeable.

Author: Ah, yes, now it's there. Thanks @aburdenthehand

2. **Enabling Workload Disruption**:
   - Set `AllowWorkloadDisruption` to `true` to allow the migration controller to act after the `AcceptableCompletionTime` threshold.

>    - To enable post-copy, set `AllowPostCopy` to `true`. Otherwise, if paused mode is prefered, unset `AllowPostCopy` or explicitly set it to `false`:

Reviewer 2: ```suggestion
   - To enable post-copy, set `AllowPostCopy` to `true`. Otherwise, if paused mode is preferred, unset `AllowPostCopy` or explicitly set it to `false`:
```
